Sources/Public/Animation/LottieAnimationLayer.swift

```diff
@@ -0,0 +1,1292 @@
+      return lhsAmount == rhsAmount
+    case (.playOnce, .playOnce),
+      (.loop, .loop),
+      (.autoReverse, .autoReverse):
+      return true
+    default:
+      return false
+    }
+  }
+}
+
+// MARK: - LottieAnimationLayer
+
+/// This class will serve as an intermediate layer between a CoreAnimationLayer
+/// and a LottieAnimationView. The motivation behind creating this class is to
+/// have all the functionality of a LottieAnimationView on a CALayer-backed object.
+
+public class LottieAnimationLayer: CALayer {
+
+  // MARK: Lifecycle
+
+  // MARK: - Public (Initializers)
+
+  /// Initializes an AnimationView with an animation.
+  public init(
+    animation: LottieAnimation?,
+    imageProvider: AnimationImageProvider? = nil,
+    textProvider: AnimationTextProvider = DefaultTextProvider(),
+    fontProvider: AnimationFontProvider = DefaultFontProvider(),
+    configuration: LottieConfiguration = .shared,
+    logger: LottieLogger = .shared)
+  {
+    self.animation = animation
+    self.imageProvider = imageProvider ?? BundleImageProvider(bundle: Bundle.main, searchPath: nil)
+    self.textProvider = textProvider
+    self.fontProvider = fontProvider
+    self.configuration = configuration
+    self.screenScale = 1
+    self.logger = logger
+    super.init()
+    makeAnimationLayer(usingEngine: configuration.renderingEngine)
+    if let animation = animation {
+      frame = animation.bounds
+    }
+  }
+
+  /// Initializes an AnimationView with a .lottie file.
+  public init(
+    dotLottie: DotLottieFile?,
+    animationId: String? = nil,
+    textProvider: AnimationTextProvider = DefaultTextProvider(),
+    fontProvider: AnimationFontProvider = DefaultFontProvider(),
+    configuration: LottieConfiguration = .shared,
+    logger: LottieLogger = .shared)
+  {
+    let dotLottieAnimation = dotLottie?.animation(for: animationId)
+    animation = dotLottieAnimation?.animation
+    imageProvider = dotLottie?.imageProvider ?? BundleImageProvider(bundle: Bundle.main, searchPath: nil)
+    self.textProvider = textProvider
+    self.fontProvider = fontProvider
+    self.configuration = configuration
+    self.screenScale = 1
+    self.logger = logger
+    super.init()
+    loopMode = dotLottieAnimation?.configuration.loopMode ?? .playOnce
+    animationSpeed = CGFloat(dotLottieAnimation?.configuration.speed ?? 1)
+    makeAnimationLayer(usingEngine: configuration.renderingEngine)
+    if let animation = animation {
+      frame = animation.bounds
+    }
+  }
+
+  public init(
+    configuration: LottieConfiguration = .shared,
+    logger: LottieLogger = .shared)
+  {
+    animation = nil
+    imageProvider = BundleImageProvider(bundle: Bundle.main, searchPath: nil)
+    textProvider = DefaultTextProvider()
+    fontProvider = DefaultFontProvider()
+    self.configuration = configuration
+    self.screenScale = 1
+    self.logger = logger
+    super.init()
+  }
+
+  required public override init(layer: Any) {
```

Comment:
This is for creating shadow `CALayer`s so should follow the patterns used elsewhere https://github.com/airbnb/lottie-ios/blob/b68c70572610afd14f013b5bd6473ac6f3fdaca2/Sources/Private/CoreAnimation/Layers/BaseCompositionLayer.swift#L29

Reply:
nice. good to know